jrnl/jrnl.py

```diff
@@ -169,6 +172,16 @@ def search_mode(args, journal, **kwargs):
     # Where do the search results go?
     if args.edit:
         _edit_search_results(**kwargs)
+        # If we want to both edit and change time in one action
+        if args.change_time:
```

Comment:
This is a good addition but I have one slight change to request. If a user runs `--edit` and `--change-time` together, then `--change-time` should happen first, and the updated timestamps should be loaded into the editor.

This avoids the awkward use case of jrnl overwriting a user's manual changes from the editor after they save the file.

Reply:
@wren I actually had it that way first, but changed it because after we change the entries' dates, they may not be selected again by `_search_journal`. For example, if the user ran `jrnl -on today --change-time tomorrow --edit`, today's entries would be changed to tomorrow, and then rerunning `_search_journal` before the call to `_edit_search_results` would select nothing.

However, while writing this I realize that we can do something to save which entries were selected and pass them on to edit, maybe apart from `_search_journal`. I'll see what I can do.

Reply:
@wren , the change has been implemented!